### DISCUSSION TITLE
Usable in bare RN projects?

### DISCUSSION THREAD

**Discussion by pvinis** [expo/router#114] [Q&A]
The new expo-router is pretty awesome. Is it possible to use it on bare RN projects?

**Comment by EvanBacon** [comment]
An Expo project is effectively a bare RN project with the `expo` package installed, it's pretty straightforward to add. Most features work on their own, e.g. Expo Modules API, Expo CLI, EAS Build, web support. Expo Router however effectively combines (or will combine) all Expo features together. Expo CLI will be used to pre-render and export routes, EAS will be used to code-sign native for universal links, some upcoming Expo Router native modules will be built on Expo Modules API, the auto splash screen is built on expo-splash-screen, the entire library is built on our React Navigation suite, testing universal links requires ngrok (which is available with `npx expo start --tunnel`), etc.

So the basics should work, but your mileage may vary for the advanced features.

**Reply by damassi** [reply]
It would be fantastic to diff what `expo-router` needs against what `expo` the package provides, and then provide a hook that connects the two so that one could use `expo-router` outside of an expo project (or with, taking advantage of the future features @EvanBacon describes above). The router as it currently stands doesn't seem to need anything related to expo, and would be an amazing project on its own.
